### PR TITLE
2 to 3

### PR DESCRIPTION
**Pull request by adlarkin**
# ➡️ Forward port

Port `ign-cmake2` to `main`

Branch comparison: https://github.com/ignitionrobotics/ign-cmake/compare/main...ign-cmake2

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by adlarkin** [comment]
I'm not sure what's going on here, but the history on the PR appears to differ from my local history. Here's what I see when I run `git diff`:
```
$ git log --oneline 
c7635f1 (HEAD -> adlarkin/2_to_3, origin/adlarkin/2_to_3) 2 to 3
3b78702 (origin/ign-cmake2, origin/adlarkin/remove_codecov_badge, origin/HEAD, ign-cmake2, adlarkin/remove_codecov_badge) Remove codecov badge from README (#172)
2f69ac3 Port codecov to new configuration (#170)
44a699c (tag: ignition-cmake2_2.8.0) 2.8.0 (#167)
d8cd96c Run ign-cmake's copy of check_test_ran (#168)
17b7058 User-friendly skip component warning (#165)
c34873d Fix hardcoded pkg-config library in examples (#163)
a1fdf09 (tag: ignition-cmake2_2.7.0) 🎈 2.7.0 (#162)
338c7f4 (tag: ignition-cmake2_2.7.0-pre2) 🎈 2.7.0~pre2 (#161)
d71927a Support to find Ogre 2-2 (#157)
01039be glib fix for Windows (#154)
a8a1f35 Master branch updates (#160)
05d8e2a (origin/main, main) 2 ➡️ 3 (#156)
```

My local history seems to match the [branch comparison](https://github.com/ignitionrobotics/ign-cmake/compare/main...ign-cmake2), but the history in this PR does not. Any ideas?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,10 +11,10 @@ jobs:
         uses: actions/checkout@v2
       - name: Compile and test
         id: ci
-        uses: ignition-tooling/action-ignition-ci@master
+        uses: ignition-tooling/action-ignition-ci@bionic
```
Comment (j-rivero) [inline .github/workflows/ci.yml:14]: do we want to use bionic on main or keeping  using main/mater from actions-ignition-ci?

Comment (chapulina) [reply .github/workflows/ci.yml:14]: We shouldn't use the `master` branch of that repository, we should delete that branch once no more libraries are using it.

The branches to be used are `bionic` and `focal`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -20,7 +20,7 @@ include(IgnCMake)
 
 #--------------------------------------
 # Set up the project
-ign_configure_project(VERSION_SUFFIX pre1)
+ign_configure_project(VERSION_SUFFIX)
```
Comment (chapulina) [inline CMakeLists.txt:23]: We shouldn't remove the `pre1` from unstable branches

File: tools/doc_check.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # that there are no doxygen warnings. For example, if you're using bitbucket
 # pipelines then you can add the following line to your bitbucket-pipelines.yml
 # file:
-#   - bash <(curl -s https://github.com/ignitionrobotics/ign-cmake/raw/master/tools/doc_check.sh)
+#   - bash <(curl -s https://github.com/ignitionrobotics/ign-cmake/raw/ign-cmake2/tools/doc_check.sh)
```
Comment (chapulina) [inline tools/doc_check.sh:7]: Nit: this could be `main`

**Review by chapulina — APPROVED** [review]

**Comment by adlarkin** [comment]
I resolved a few conflicts related to #169. As discussed in https://github.com/ignitionrobotics/ign-cmake/pull/170#issuecomment-841413236, we are disabling `codecov` in this repository, and as mentioned in https://github.com/ignitionrobotics/ign-cmake/pull/173#discussion_r632801478, we should be using either the `boinic` or `focal` branches for `actions-ignition-ci`.

@j-rivero, mind taking one last look at this before I merge?

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**